scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,166 @@ local function enforceRuneCounts(target)
     end
 end
 
+--source https://www.bluegartr.com/threads/124844-Vivacious-Pulse-testing?p=6038534&viewfull=1#post6038534 and following posts
+local function getRuneHealAmount(effect, target)
+
+    if effect == xi.effect.TENEBRAE then -- Tenebrae primarily restores MP, but also uses the base divine skill/2 bonus but no raw stat bonus.
+        return 0
+    end
+
+    local runeStatMap = {
+                            [xi.effect.IGNIS]  = xi.mod.STR,
+                            [xi.effect.GELUS]  = xi.mod.DEX,
+                            [xi.effect.FLABRA] = xi.mod.VIT,
+                            [xi.effect.TELLUS] = xi.mod.AGI,
+                            [xi.effect.SULPOR] = xi.mod.INT,
+                            [xi.effect.UNDA]   = xi.mod.MND,
+                            [xi.effect.LUX]    = xi.mod.CHR,
+                        }
+
+    local stat = runeStatMap[effect:getType()]
+    if stat ~= nil then
+        return math.floor(target:getStat(stat) * 0.5)
+    end
+
+    return 0
+end
+
+
+
+--source https://www.bg-wiki.com/ffxi/Vivacious_Pulse
+local function calculateVivaciousPulseHealing(target)
+
+    local divineMagicSkillLevel = target:getSkillLevel(xi.skill.DIVINE_MAGIC)
+    local HPHealAmount = 10 + math.floor(divineMagicSkillLevel / 2 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) -- Bonus of 1-20%  from Vivacious pulse job points.
+    local tenebraeRuneCount = 0
+    local bonusPct = 1.0 -- todo, add in bonus from augment attached to Futhark Claymore, Peord Claymore, and Morgelai. Need extdata for proper values.
+
+    local debuffs = {}
+    local debuffCount = 0
+
+    local effects = target:getStatusEffects()
+    for _, effect in ipairs(effects) do
+        local type = effect:getType()
+        if type >= xi.effect.IGNIS and type <= xi.effect.TENEBRAE then
+            HPHealAmount = HPHealAmount + getRuneHealAmount(effect, target)
+	    elseif type == xi.effect.POISON or type == xi.effect.PARALYSIS or type == xi.effect.BLINDNESS
+            or type == xi.effect.SILENCE or type == xi.effect.MUTE or type == xi.effect.CURSE_I
+            or type == xi.effect.CURSE_II or type == xi.effect.DOOM or type == xi.effect.VIRUS
+            or type == xi.effect.PLAGUE or type == xi.effect.PETRIFICATION then
+             debuffs[debuffCount+1] = type
+             debuffCount = debuffCount + 1
```

Comment:
The elseif isn't properly aligned. (Might be just in my end though, please confirm, just in case.)
Also, we normaly leave the "and" / "or" at the end of the line, like...
```
    elseif
        type == xi.effect.POISON or type == xi.effect.PARALYSIS or type == xi.effect.BLINDNESS or
        type == xi.effect.SILENCE or type == xi.effect.MUTE or type == xi.effect.CURSE_I or
        type == xi.effect.CURSE_II or type == xi.effect.DOOM or type == xi.effect.VIRUS or
        type == xi.effect.PLAGUE or type == xi.effect.PETRIFICATION
    then
```

Reply:
Thanks for the tip on the styling there. I couldn't decide how I wanted it but now that I see that it makes sense.